sdk/src/cache/messages/data/scalar/get_batch.rs

```diff
@@ -0,0 +1,224 @@
+///     }
+/// }
+/// # Ok(())
+/// # })
+/// # }
+/// ```
+pub struct GetBatchRequest<K: IntoBytesIterable> {
+    cache_name: String,
+    keys: K,
+}
+
+impl<K: IntoBytesIterable> GetBatchRequest<K> {
+    /// Constructs a new GetBatchRequest.
+    pub fn new(cache_name: impl Into<String>, keys: K) -> Self {
+        Self {
+            cache_name: cache_name.into(),
+            keys,
+        }
+    }
+}
+
+impl<K: IntoBytesIterable> MomentoRequest for GetBatchRequest<K> {
+    type Response = GetBatchResponse;
+
+    async fn send(self, cache_client: &CacheClient) -> MomentoResult<GetBatchResponse> {
+        // consume self.keys once to convert all keys to bytes
+        let byte_keys: Vec<Vec<u8>> = self.keys.into_bytes();
+
+        // convert keys to strings for the HashMap keys
+        let string_keys: Vec<String> = byte_keys
+            .iter()
+            .map(|key| parse_string(key.clone()))
+            .collect::<MomentoResult<Vec<String>>>()?;
+
+        let get_requests = byte_keys
+            .iter()
+            .map(|key| momento_protos::cache_client::GetRequest {
+                cache_key: key.clone(),
+            })
+            .collect();
+
+        let get_batch_request = utils::prep_request_with_timeout(
+            &self.cache_name,
+            cache_client.deadline_millis(),
+            momento_protos::cache_client::GetBatchRequest {
+                items: get_requests,
+            },
+        )?;
+
+        let mut response_stream = cache_client
+            .next_data_client()
+            .get_batch(get_batch_request)
+            .await?
+            .into_inner();
+
+        // receive stream of get responses
+        // let mut results_list: Vec<GetResponse> = vec![];
+        let mut responses: HashMap<String, GetResponse> = HashMap::new();
+        let mut string_keys_iter = string_keys.iter();
+        while let Some(get_response) = response_stream.message().await? {
+            let sdk_get_response = match get_response.result() {
+                ECacheResult::Hit => GetResponse::Hit {
+                    value: Value {
+                        raw_item: get_response.cache_body,
+                    },
+                },
+                ECacheResult::Miss => GetResponse::Miss,
+                _ => {
+                    return Err(MomentoError::unknown_error(
+                        "GetBatch",
+                        Some(format!("{:#?}", get_response)),
+                    ))
+                }
+            };
+            let key = match string_keys_iter.next() {
+                Some(key) => key,
+                None => {
+                    return Err(MomentoError::unknown_error(
+                        "GetBatch",
+                        Some("Received more responses than expected".to_string()),
+                    ))
+                }
+            };
+            responses.insert(key.to_string(), sdk_get_response);
```

Comment:
similarly you do not need to memcpy each of the keys again:
```suggestion
        let mut string_keys_iter = string_keys.into_iter();
        while let Some(get_response) = response_stream.message().await? {
            let sdk_get_response = match get_response.result() {
                ECacheResult::Hit => GetResponse::Hit {
                    value: Value {
                        raw_item: get_response.cache_body,
                    },
                },
                ECacheResult::Miss => GetResponse::Miss,
                _ => {
                    return Err(MomentoError::unknown_error(
                        "GetBatch",
                        Some(format!("{:#?}", get_response)),
                    ))
                }
            };
            let key = match string_keys_iter.next() {
                Some(key) => key,
                None => {
                    return Err(MomentoError::unknown_error(
                        "GetBatch",
                        Some("Received more responses than expected".to_string()),
                    ))
                }
            };
            responses.insert(key, sdk_get_response);
```